halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java

```diff
@@ -73,6 +83,50 @@ protected void setProfile(
     profile.appendContents("pipelineTemplates.enabled: " + pipelineTemplates);
     // For backward compatibility
     profile.appendContents("pipelineTemplate.enabled: " + pipelineTemplates);
+
+    // Loading Plugin Information
+    Yaml yaml = new Yaml();
+    final List<Plugin> plugins = deploymentConfiguration.getPlugins().getPlugin();
+    Map<String, Object> fullyRenderedYaml = new LinkedHashMap<>();
+    Map<String, Object> pluginMetadata = new LinkedHashMap<>();
+
+    for (Plugin plugin : plugins) {
+      if (!plugin.getEnabled()) {
+        log.info("Plugin " + plugin.getName() + ", not enabled");
+        continue;
+      }
+
+      String manifestLocation = plugin.getManifestLocation();
+      if (Objects.equals(manifestLocation, null)) {
+        log.info("Plugin " + plugin.getName() + ", has no manifest file");
+        continue;
+      }
+
+      InputStream manifestContents;
+      try {
+        if (manifestLocation.startsWith("http:") || manifestLocation.startsWith("https:")) {
+          URL url = new URL(manifestLocation);
+          manifestContents = url.openStream();
+        } else {
+          manifestContents = new FileInputStream(manifestLocation);
+        }
+
+        Map<String, Object> manifest = yaml.load(manifestContents);
+        String pluginName = (String) manifest.get("name");
```

Comment:
As discussed on Zoom, I think adding a manifest version so that we can handle backwards breaking changes in manifest structure would be good. In future we'll be free to make those kinds of changes without breaking any future plugins on an old version.

Reply:
added 👍